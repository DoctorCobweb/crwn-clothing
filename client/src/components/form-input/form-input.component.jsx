import React from 'react';

import {
  GroupContainer,
  FormInputContainer,
  FormInputLabel
} from './form-input.styles.jsx';

export const FormInput = ({ handleChange, label, ...otherProps }) => (
  <GroupContainer>
    <FormInputContainer onChange={handleChange} {...otherProps} />
    {
      label ?
      (<FormInputLabel textLength={otherProps.value.length}>
        {label}
      </FormInputLabel>)
      : null
    }
  </GroupContainer>
);

export default FormInput;