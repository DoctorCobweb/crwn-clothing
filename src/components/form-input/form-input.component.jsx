import React from 'react';

import {
  GroupContainer,
  FormInputContainer,
  FormInputLabel
} from './form-input.styles.jsx';
// import './form-input.styles.scss';

const FormInput = ({ handleChange, label, ...otherProps }) => (
  <GroupContainer>
    <FormInputContainer onChange={handleChange} {...otherProps} />
    {
      label ?
      (<FormInputLabel textLength={otherProps.value.length}>
        {label}
      </FormInputLabel>)
      : null
    }
  </GroupContainer>
);

// const FormInput = ({ handleChange, label, ...otherProps }) => (
//   <div className="group">
//     <input className="form-input" onChange={handleChange} {...otherProps} />
//     {
//       label ?
//       (<label className={`${otherProps.value.length ? 'shrink': ''} form-input-label`}>
//         {label}
//       </label>)
//       : null
//     }
//   </div>

// );

export default FormInput;