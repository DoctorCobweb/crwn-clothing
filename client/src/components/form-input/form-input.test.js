import React from 'react';
import { shallow } from 'enzyme';

import { FormInput } from './form-input.component';

describe('FormInput component', ()=>{
  let wrapper;
  let mockProps;
  let mockHandleChange;

  beforeEach(()=>{
    mockHandleChange = jest.fn();
    mockProps = {
      handleChange: mockHandleChange,
      label: 'email',
      value: 'test@gmail.com',
    }
    wrapper = shallow(<FormInput {...mockProps} />);
  })

  it('should render FormInput', ()=> {
    expect(wrapper.debug()).toMatchSnapshot();
  })

  it('should call handleChange when input changes', ()=>{
    wrapper.find('FormInputContainer').simulate('change');
    expect(mockProps.handleChange).toHaveBeenCalled();
  });

  it('should render FormInputLabel component for non-null label prop val', ()=>{
    // this is shorthand for the below commented-out line
    expect(wrapper.exists('FormInputLabel')).toBe(true);
    // expect(wrapper.find('FormInputLabel').exists()).toBe(true);
  });

  it('should not render FormInputLabel component if there is no label', ()=>{
    const newMockProps = {
      handleChange: mockHandleChange,
      label: '',
      value: 'test@gmail.com',
    };
    const newWrapper = shallow(<FormInput {...newMockProps} />);
    expect(newWrapper.exists('FormInputLabel')).toBe(false);
  });
})