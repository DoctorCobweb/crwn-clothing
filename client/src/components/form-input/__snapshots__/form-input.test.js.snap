// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component should render FormInput 1`] = `
"<GroupContainer>
  <FormInputContainer onChange={[Function: mockConstructor]} value=\\"test@gmail.com\\" />
  <FormInputLabel textLength={14}>
    email
  </FormInputLabel>
</GroupContainer>"
`;
